src/drivers/serac.cpp

```diff
@@ -80,33 +80,44 @@ int main(int argc, char* argv[])
     input_file_path = search->second;
   }
 
+  // Check if a restart was requested
+  std::optional<int> restart_cycle;
+  if (auto cycle = cli_opts.find("restart_cycle"); cycle != cli_opts.end()) {
+    restart_cycle = std::stoi(cycle->second);
+  }
+
   // Check for the doc creation command line argument
   bool create_input_file_docs = cli_opts.find("create_input_file_docs") != cli_opts.end();
 
   // Create DataStore
   axom::sidre::DataStore datastore;
 
+  // Intialize MFEMSidreDataCollection
+  serac::StateManager::initialize(datastore, restart_cycle);
+
   // Initialize Inlet and read input file
   auto inlet = serac::input::initialize(datastore, input_file_path);
   serac::defineInputFileSchema(inlet, rank);
 
   // Optionally, create input file documentation and quit
   if (create_input_file_docs) {
-    auto writer = std::make_unique<axom::inlet::SphinxDocWriter>("serac_input.rst", inlet.sidreGroup());
-    inlet.registerDocWriter(std::move(writer));
+    auto writer = std::make_unique<axom::inlet::SphinxWriter>("serac_input.rst");
+    inlet.registerWriter(std::move(writer));
     inlet.writeDoc();
     serac::exitGracefully();
   }
 
   // Save input values to file
   datastore.getRoot()->save("serac_input.json", "json");
```

Comment:
This originally was "hacked" in to show the input values in a quick and dirty way.  This PR will add extra values to the sidre tree on restarts.  Can you either move this up or probably better yet, only print out the input part of the tree instead of the whole root.